pkg/validations/compare.go

```diff
@@ -24,21 +24,49 @@ import (
 	"sigs.k8s.io/crdify/pkg/config"
 )
 
+// PropertyComparisonResult represents the results
+// from running validations against a given property of
+// a CustomResourceDefinition.
+type PropertyComparisonResult struct {
+	// property is the property that the
+	// set of compatibility validation results
+	// applies to.
+	Property string `json:"property"`
```

Comment:
Worth adding that this is considered a json simple path? (It is json simple path right?)